ffs/scheduler/internal/sjstore/sjstore.go

```diff
@@ -51,18 +59,59 @@ type watcher struct {
 // New returns a new JobStore backed by the Datastore.
 func New(ds datastore.Datastore) (*Store, error) {
 	s := &Store{
-		ds:            ds,
-		executingCids: make(map[cid.Cid]ffs.JobID),
+		ds:                 ds,
+		executingCids:      make(map[cid.Cid]ffs.JobID),
+		jobStatusCache:     make(map[ffs.APIID]map[cid.Cid]map[cid.Cid]deals.StorageDealInfo),
+		queuedJobs:         make(map[ffs.APIID]map[cid.Cid][]*ffs.StorageJob),
+		executingJobs:      make(map[ffs.APIID]map[cid.Cid]*ffs.StorageJob),
+		lastFinalJobs:      make(map[ffs.APIID]map[cid.Cid]*ffs.StorageJob),
+		lastSuccessfulJobs: make(map[ffs.APIID]map[cid.Cid]*ffs.StorageJob),
 	}
-	if err := s.loadExecutingJobs(); err != nil {
-		return nil, fmt.Errorf("reloading priori executing jobs: %s", err)
-	}
-	if err := s.loadQueuedJobs(); err != nil {
-		return nil, fmt.Errorf("loading pending jobs: %s", err)
+	if err := s.loadCaches(); err != nil {
+		return nil, fmt.Errorf("reloading caches: %s", err)
 	}
 	return s, nil
 }
 
+// MonitorJob returns a channel that can be passed into the deal monitoring process.
+func (s *Store) MonitorJob(j ffs.StorageJob) chan deals.StorageDealInfo {
+	dealUpdates := make(chan deals.StorageDealInfo, 1000)
+	go func() {
+		for update := range dealUpdates {
+			s.lock.Lock()
+			_, ok := s.jobStatusCache[j.APIID]
+			if !ok {
+				s.jobStatusCache[j.APIID] = map[cid.Cid]map[cid.Cid]deals.StorageDealInfo{}
+			}
+			_, ok = s.jobStatusCache[j.APIID][j.Cid]
+			if !ok {
+				s.jobStatusCache[j.APIID][j.Cid] = map[cid.Cid]deals.StorageDealInfo{}
+			}
+			s.jobStatusCache[j.APIID][j.Cid][update.ProposalCid] = update
+			job, err := s.get(j.ID)
+			if err != nil {
+				log.Errorf("getting job: %v", err)
```

Comment:
I think you're assuming I had a reason for this :)

Just an oversight, will  add `continue`, that is the correct thing.

Reply:
went with this just to be safe:
```
			job, err := s.get(j.ID)
			if err == ErrNotFound {
				log.Errorf("job not found: %v", err)
				s.lock.Unlock()
				return
			}
			if err != nil {
				log.Errorf("getting job: %v", err)
				s.lock.Unlock()
				continue
			}
```

Reply:
Uhm, yeah tricky to `return`. The usual problem with returning a channel that can indicate that something bad happens... so the caller can't distinguish between "all is done and good, just finished" from "this was closed earlier since an error happened".

Reply:
How bad that situation of misinterpreting a "success" call from an "errored" call be for clients that are using this API?

Reply:
Ah yea. In this case, I'd say it isn't very important to convey that error to listeners. Removed the `return` and just keeping `continue`. 

The possible error is very unlikely, or impossible, and if it does happen, the worse case is that listeners don't get called back with updates. The error is logged, so we'll have something to look for if a user reports not receiving job updates.